src/l500/l500-options.cpp

```diff
@@ -36,23 +37,59 @@ namespace librealsense
         return _range;
     }
 
+    float l500_hw_options::query_default( hwmon_response * response ) const
+    {
+        if(_fw_version >= firmware_version( MIN_GET_DEFAULT_FW_VERSION ) )
+        {
+            return get_default( int( _resolution->query() ), response );
+        }
+        else
+        {
+            // For older FW without support for get_default, we have to:
+            //     1. Save the current value
+            //     2. Change to -1
+            //     3. Wait for the next frame (if streaming)
+            //     4. Read the current value
+            //     5. Restore the current value
+            auto current = query_current( ( rs2_sensor_mode )( int( _resolution->query() ) ) );
```

Comment:
I suggest, since this piece of code keeps repeating, a `query_sensor_mode()` function

Reply:
done